service/src/main/java/tds/config/repositories/impl/ConfigRepositoryImpl.java

```diff
@@ -58,4 +65,128 @@ public ConfigRepositoryImpl(NamedParameterJdbcTemplate jdbcTemplate) {
 
         return clientSystemFlags;
     }
+
+    @Override
+    public Optional<ClientSystemMessage> findClientSystemMessage(String clientName, String messageKey, String language, String clientDefaultLanguage, String context, String subject, String grade) {
+        List<String> langauges = Arrays.asList(language, clientDefaultLanguage);
+        List<String> grades = new ArrayList<>(Arrays.asList(MATCH_ANY));
+        List<String> subjects = new ArrayList<>(Arrays.asList(MATCH_ANY));
+
+        if (grade != null && !grade.equals(MATCH_ANY)) {
```

Comment:
Turns out this won't work because we are checking not equals, and `null` and `MATCH_ANY` are not equal but we don't want that added to the list